src/umpire/resource/DefaultMemoryResource.inl

```diff
@@ -55,8 +55,8 @@ void DefaultMemoryResource<_allocator>::deallocate(void* ptr)
 
   UMPIRE_RECORD_STATISTIC(getName(), "ptr", reinterpret_cast<uintptr_t>(ptr), "size", 0x0, "event", "deallocate");
 
-  m_allocator.deallocate(ptr);
   deregisterAllocation(ptr);
+  m_allocator.deallocate(ptr);
```

Comment:
I think that reordering this is the correct thing to do here.  Without the reordering, the device will be given the reallocation before we even look to see it is something we have registered.